x/distribution/keeper/msg_server.go

```diff
@@ -32,6 +32,12 @@ func (k msgServer) SetWithdrawAddress(ctx context.Context, msg *types.MsgSetWith
 		return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid delegator address: %s", err)
 	}
 
+	acc := k.authKeeper.GetAccount(ctx, delegatorAddress)
+	_, ok := acc.(types.VestingAccount)
+	if ok {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("cannot set withdraw address for vesting account %s", delegatorAddress)
+	}
```

Comment:
[question] would it ever make sense for a vesting account to set the withdraw address as the original vesting account address? If yes, should we permit that case? The scenario I'm thinking of:

1. There's a vesting account on Mainnet already that set withdraw address to X.
2. CIP-31 activates on Mainnet
3. The vesting account attempts to modify it's withdraw address to it's own address but hits this error.

It's prob fine to keep this as-is because even if the vesting account hits the "cannot set withdraw address for vesting account" error, rewards will only be withdrawn to the vesting account.